src/com/facebook/buck/file/HttpDownloader.java

```diff
@@ -57,6 +68,19 @@ public boolean fetch(BuckEventBus eventBus, URI uri, Path output) throws IOExcep
 
     try {
       HttpURLConnection connection = createConnection(uri);
+
+      if (authentication.isPresent()) {
+        if ("https".equals(uri.getScheme())) {
```

Comment:
Also check that the connections is an instance of https://docs.oracle.com/javase/7/docs/api/javax/net/ssl/HttpsURLConnection.html .